influxdb3_write/src/paths.rs

```diff
@@ -0,0 +1,128 @@
+use crate::SegmentId;
+use std::convert::AsRef;
+use std::ops::Deref;
+use std::path::Path;
+use std::path::PathBuf;
+
+/// File extension for catalog files
+const CATALOG_FILE_EXTENSION: &str = "json";
+
+/// File extension for parquet files
+const PARQUET_FILE_EXTENSION: &str = "parquet";
+
+/// File extension for segment files
+const SEGMENT_FILE_EXTENSION: &str = "wal";
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct CatalogFilePath(PathBuf);
+
+impl CatalogFilePath {
+    pub fn new(prefix: impl Into<PathBuf>, sequence_number: u64) -> Self {
+        let mut path = prefix.into();
+        path.push("catalogs");
+        path.push(format!("{sequence_number:010}"));
```

Comment:
I opted out of this for wal files since they don't really need to be human readable, but for the other types I did the `u32::MAX` trick so that they show up in order.